pkg/sql/parser/sql.y

```diff
@@ -4295,6 +4297,40 @@ explain_option_list:
     $$.val = append($1.strs(), $3)
   }
 
+// %Help: ALTER BACKUP - alter an existing backup's encryption keys
+// %Category: CCL
+// %Text:
+// ALTER BACKUP <location...>
+//        [ ADD NEW_KMS = <kms...> ]
+//        [ WITH OLD_KMS = <kms...> ]
+// Locations:
+//    "[scheme]://[host]/[path to backup]?[parameters]"
+//
+// KMS:
+//    "[kms_provider]://[kms_host]/[master_key_identifier]?[parameters]" : add new kms keys to backup
+alter_backup_stmt:
+  ALTER BACKUP list_of_string_or_placeholder_opt_list
```

Comment:
Spoke to @DarrylWong offline, I think we should be doing something similar to what `SHOW BACKUP` does and only accept a single backup directory that points all the way to the `ENCRYPTION-INFO` file. In that case this should change to `string_or_placeholder`.